core/pom.xml

```diff
@@ -66,11 +66,23 @@ THE SOFTWARE.
     <dependency>
       <groupId>${project.groupId}</groupId>
       <artifactId>remoting</artifactId>
+      <exclusions>
+        <exclusion>
+          <groupId>org.jenkins-ci</groupId>
+          <artifactId>constant-pool-scanner</artifactId>
```

Comment:
Used inside the remoting: https://github.com/jenkinsci/remoting/blob/9099602f9d4b5442ff3b8624c1aad29d9dca1442/src/main/java/hudson/remoting/RemoteClassLoader.java  . It might be unused inside the Jenkins core paths by default and by downstream plugins, but it is impossible to say for sure

Reply:
`constant-pool-scanner` is an old library. It might not detect some modern Java references in the constant pool, but I agree with you that it is not impossible to know for sure where it is used.